src/components/ledger/LedgerModal.tsx

```diff
@@ -92,24 +95,40 @@ function LedgerModal({
   }
 
   return (
-    <Modal
-      hasCloseButton
-      isSlideUp
-      isOpen={isOpen}
-      onClose={onClose}
-      onCloseAnimationEnd={handleLedgerModalClose}
-      dialogClassName={styles.modalDialog}
-    >
-      <Transition
-        name="pushSlide"
-        className={buildClassName(modalStyles.transition, 'custom-scroll')}
-        slideClassName={modalStyles.transitionSlide}
-        activeKey={currentSlide}
-        nextKey={nextKey}
-      >
-        {renderContent}
-      </Transition>
-    </Modal>
+    <div>
+      {isInsideModal ? (
+        <Transition
+          name="pushSlide"
+          className={buildClassName(modalStyles.transition, 'custom-scroll')}
+          slideClassName={modalStyles.transitionSlide}
+          activeKey={currentSlide}
+          nextKey={nextKey}
+        >
+          {renderContent}
+        </Transition>
+      ) : (
+        <Modal
+          hasCloseButton
+          isSlideUp
+          isOpen={isOpen}
+          onClose={onClose}
+          onCloseAnimationEnd={handleLedgerModalClose}
+          dialogClassName={styles.modalDialog}
+        >
+          <Transition
+            name="pushSlide"
+            className={buildClassName(modalStyles.transition, 'custom-scroll')}
+            slideClassName={modalStyles.transitionSlide}
+            activeKey={currentSlide}
+            nextKey={nextKey}
+          >
+            {renderContent}
+          </Transition>
```

Comment:
Maybe it would be better to put this code in a function, so as not to repeat it on lines 100-108?

And in src/components/main/modals/BackupModal.tsx file the same suggestion is made.

Reply:
I meant to take out the Transition component specifically.

In pseudo-code, it would be like this:

```
function renderContentTransition() {
  return <Transition ... />
}

if (isInsideModal) {
  return renderContentTransition();
}

return (
  <Modal ...>
    {renderContentTransition()}
  </Modal>
);
```

Can we do it this way?

Reply:
Yeah, that's a great idea.